src/pages/member/UserEvents.js

```diff
@@ -0,0 +1,292 @@
+    if (eventsRegistered.length && typeof eventsRegistered[0] === 'object') {
+      return eventsRegistered.map((event) => event.eventID)
+    }
+
+    return eventsRegistered
+  }, [eventsRegistered])
+
+  const eventsFavoritedIds = useMemo(() => {
+    if (user && user.favedEventsID && user.favedEventsID.length) {
+      return user.favedEventsID
+    }
+
+    return []
+  }, [user.favedEventsID])
+
+  const AllEventCards = useMemo(() => {
+    if (selectedPanel === PANEL_STATES.REGISTERED) return events.filter((event) => eventsRegisteredIds.includes(event.id))
+    else if (selectedPanel === PANEL_STATES.FAVORITES) return events.filter((event) => eventsFavoritedIds.includes(event.id))
+    else return events
+  }, [events, selectedPanel, user.favedEventsID])
+
+  const UpcomingEventCards = useMemo(() => {
+    // filter events by the date
+    const now = new Date()
+    return AllEventCards.filter((event) => event.startDate && new Date(event.startDate) >= now)
+  }, [AllEventCards, selectedPanel])
+
+  const PastEventCards = useMemo(() => {
+    // filter events by the date
+    const now = new Date()
+    return AllEventCards.filter((event) => event.startDate && new Date(event.startDate) < now)
+  }, [AllEventCards, selectedPanel])
+
+  return (
+    <div>
+      <Helmet>
+        <title>Biztech User Events Dashboard</title>
+      </Helmet>
+      <div style={styles.flex}>
+        <div style={styles.sidePanel.layout}>
+          <Typography variant='h1' style={styles.header}>Events</Typography>
+          <List>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.FAVORITES
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.FAVORITES)}
+              button
+            >
+              <ListItemText><StarBorder fontSize='small' />&nbsp;Favorites</ListItemText>
+            </ListItem>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.REGISTERED
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.REGISTERED)}
+              button
+            >
+              <ListItemText><PlaylistAddCheck fontSize='small' />&nbsp;Registered</ListItemText>
+            </ListItem>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.ALL
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.ALL)}
+              button
+            >
+              <ListItemText><Search fontSize='small' />&nbsp;All</ListItemText>
+            </ListItem>
+          </List>
+        </div>
+        <div style={styles.tabs.container}>
+
+          <Tabs
+            value={tabIndex}
+            indicatorColor='primary'
+            textColor='primary' // TODO: Set the primary color theme
```

Comment:
can we set the primary colour now?

Reply:
Oh ya Derek's PR helped solve this [here](https://github.com/ubc-biztech/bt-web/blob/7cbc49779a1a27e1b168b73051a938a753cf9404/src/components/ThemeProvider.js#L62-L64) 👍